PowerEditor/src/WinControls/Grid/ShortcutMapper.cpp

```diff
@@ -352,6 +357,120 @@ INT_PTR CALLBACK ShortcutMapper::run_dlgProc(UINT message, WPARAM wParam, LPARAM
+							//Get CommandShortcut corresponding to row
+							vector<CommandShortcut> & shortcuts = nppParam->getUserShortcuts();
+							CommandShortcut csc = shortcuts[row - 1];
+							csc.clear();
+							shortcuts[row - 1] = csc;
+							//shortcut was altered
+							nppParam->addUserModifiedIndex(row-1);
+						
+							//save the current view
+							_lastHomeRow[_currentState] = _babygrid.getHomeRow();
+							_lastCursorRow[_currentState] = _babygrid.getSelectedRow();
+							fillOutBabyGrid();
+
+							isModified = true;
+
+							//Notify current Accelerator class to update everything
+							nppParam->getAccelerator()->updateShortcuts();
+							nppParam->setShortcutDirty();
+							break;
+						}
+						case STATE_MACRO: 
+						{
+							//Get MacroShortcut corresponding to row
+							vector<MacroShortcut> & shortcuts = nppParam->getMacroList();
+							MacroShortcut msc = shortcuts[row - 1];
+							msc.clear();
+							shortcuts[row - 1] = msc;
+							//save the current view
+							_lastHomeRow[_currentState] = _babygrid.getHomeRow();
+							_lastCursorRow[_currentState] = _babygrid.getSelectedRow();
+							fillOutBabyGrid();
+
+							isModified = true;
+
+							//Notify current Accelerator class to update everything
+							nppParam->getAccelerator()->updateShortcuts();
+							nppParam->setShortcutDirty();
+							break;
+						}
+						case STATE_USER: 
+						{
+							//Get UserCommand corresponding to row
+							vector<UserCommand> & shortcuts = nppParam->getUserCommandList();
+							UserCommand ucmd = shortcuts[row - 1];
+							ucmd.clear();
+						
+							//shortcut was altered
+							shortcuts[row - 1] = ucmd;
+
+							//save the current view
+							_lastHomeRow[_currentState] = _babygrid.getHomeRow();
+							_lastCursorRow[_currentState] = _babygrid.getSelectedRow();
+							fillOutBabyGrid();
+
+							isModified = true;
+
+							//Notify current Accelerator class to update everything
+							nppParam->getAccelerator()->updateShortcuts();
+							nppParam->setShortcutDirty();
+							break;
+						}
+						case STATE_PLUGIN: 
+						{
+							//Get PluginCmdShortcut corresponding to row
+							vector<PluginCmdShortcut> & shortcuts = nppParam->getPluginCommandList();
+							PluginCmdShortcut pcsc = shortcuts[row - 1];
+							pcsc.clear();
+							//shortcut was altered
+							nppParam->addPluginModifiedIndex(row-1);
+							shortcuts[row - 1] = pcsc;
+						
+							//save the current view
+							_lastHomeRow[_currentState] = _babygrid.getHomeRow();
+							_lastCursorRow[_currentState] = _babygrid.getSelectedRow();
+							fillOutBabyGrid();
+
+							isModified = true;
+
+							//Notify current Accelerator class to update everything
+							nppParam->getAccelerator()->updateShortcuts();
+							unsigned long cmdID = pcsc.getID();
+							ShortcutKey shortcut;
+							shortcut._isAlt = FALSE;
+							shortcut._isCtrl = FALSE;
+							shortcut._isShift = FALSE;
+							shortcut._key = '\0';

```

Comment:
Though technically not needed, this would be more readable as `shortcut.clear();`